internal/dataplane/kongstate/customentity.go

```diff
@@ -0,0 +1,87 @@
+package kongstate
+
+import "github.com/kong/go-kong/kong"
+
+// EntityFieldType represents type of a Kong entity field.
+// possible field types include boolean, integer, number, string, array, set, map, record, json, foreign.
+type EntityFieldType string
+
+const (
```

Comment:
The docs page is weirdly moot about the fundamental types (it links a reference to the subtypes, like `timeout` as a special class of `integer`, but nothing about the base types that I can see), but I don't think there's a better page that covers the DAO.

```suggestion
// EntityFieldType represents type of a Kong entity field.
// possible field types include boolean, integer, number, string, array, set, map, record, json, foreign.
type EntityFieldType string

// These types and field properties are defined upstream in the Kong DAO:
// https://github.com/Kong/kong/blob/3.6.1/kong/db/schema/init.lua#L131-L143
// https://docs.konghq.com/gateway/latest/plugin-development/custom-entities/#define-a-schema

const (
```